### PR TITLE
Docs: Update the doc on default hint and execution devices property

### PR DESCRIPTION
**Pull request by WeldonWangwang**
### Details:
 - *ov::execution_devices in 92659*
 - Update to LATENCY as default hint

### CODE REVIEW

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 1]
```diff
@@ -196,24 +196,24 @@ AUTO will then query all available devices and remove CPU from the candidate lis
...
 
-#### THROUGHPUT
-This option prioritizes high throughput, balancing between latency and power. It is best suited for tasks involving multiple jobs, such as inference of video feeds or large numbers of images.
-
-> **NOTE**: If no performance hint is set explicitly, AUTO will set THROUGHPUT for devices that have not set `ov::device::properties`. For example, if you have both a CPU and a GPU in the system, this command `core.compile_model("AUTO", ov::device::properties("CPU", ov::enable_profiling(true)))` will set THROUGHPUT for the GPU only. No hint will be set for the CPU although it's the selected device.
-
 #### LATENCY
 This option prioritizes low latency, providing short response time for each inference job. It performs best for tasks where inference is required for a single input image, e.g. a medical analysis of an ultrasound scan image. It also fits the tasks of real-time or nearly real-time applications, such as an industrial robot's response to actions in its environment or obstacle avoidance for autonomous vehicles.
 
+> **NOTE**: If no performance hint is set explicitly, AUTO will set LATENCY for devices that have not set `ov::device::properties`.
```
Comment (peterchen-intel) [inline docs/OV_Runtime_UG/auto_device_selection.md:206]: https://github.com/openvinotoolkit/openvino/pull/14236/files#diff-c4165554e307d3dec493e4115e22b369f59e8cd9aef2f1639b38779d5eff0becR73 @yangwang201911 If NUM_STREAMS is a "AUTO property"?

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 2]
```diff
@@ -269,6 +269,26 @@ The `ov::hint::model_priority` property enables you to control the priorities of
 
 @endsphinxdirective
 
+### Execution Device Priority
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/auto_device_selection.md:272]: What's the differences between `ov::execution_devices` and `ov::runtime::Core::get_available_devices()`? This section seems not talking about priorities, but only "query runtime target devices". Should it be put under `#### Checking Available Devices` above?

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 1 above)
Comment (peterchen-intel) [reply docs/OV_Runtime_UG/auto_device_selection.md:206]: No.

**Comment by peterchen-intel** [comment]
To master: https://github.com/openvinotoolkit/openvino/pull/14750

**Comment by WeldonWangwang** [comment]
Hi @xu-yuan1 ,
 ov::execution_devices return the name of device that actual compiled model, ov::runtime::Core::get_available_devices() return the list of available devices.
For example:
When compile model with -d AUTO:GPU.0 in a machine that has CPU and one iGPU. ov::runtime::Core::get_available_devices() will return "CPU, GPU", but ov::execution_devices will return "GPU.0".

And I moved this property description to the property table in the document.

**Comment by xu-yuan1** [comment]
> Hi @xu-yuan1 , ov::execution_devices return the name of device that actual compiled model, ov::runtime::Core::get_available_devices() return the list of available devices. For example: When compile model with -d AUTO:GPU.0 in a machine that has CPU and one iGPU. ov::runtime::Core::get_available_devices() will return "CPU, GPU", but ov::execution_devices will return "GPU.0".
> 
> And I moved this property description to the property table in the document.

Thanks for that! I think it's still good to keep a section about ov::execution_devices to describe more information. How about putting the section right before "Performance Hints" as it's about showing the target runtime devices after selection? 
If it's not about priorities, we can name it something like "### Checking Target Runtime Devices", and add a sentence or two to describe when the users may want to use it, together with an example.

**Comment by WeldonWangwang** [comment]
> Thanks for that! I think it's still good to keep a section about ov::execution_devices to describe more information. How about putting the section right before "Performance Hints" as it's about showing the target runtime devices after selection? If it's not about priorities, we can name it something like "### Checking Target Runtime Devices", and add a sentence or two to describe when the users may want to use it, together with an example.

@xu-yuan1  Added some descriptions, please review. thank you.

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 3]
```diff
@@ -195,25 +199,45 @@ AUTO will then query all available devices and remove CPU from the candidate lis
 
 Note that if you choose to exclude CPU from device candidate list, CPU will not be able to support the initial model compilation stage. See more information in [How AUTO Works](#how-auto-works).
 
-### Performance Hints for AUTO
-The `ov::hint::performance_mode` property enables you to specify a performance option for AUTO to be more efficient for particular use cases.
+### Checking Target Runtime Devices
 
-> **NOTE**: Currently, the `ov::hint` property is supported by CPU and GPU devices only.
+The `ov::execution_devices` property enables you to query runtime target devices in the Auto-Device plugin.
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/auto_device_selection.md:204]:
```suggestion
To query the runtime target devices while running a model using AUTO, you can use the `ov::execution_devices` property. It must be used with the `get_property` property, for example:
```

**Review by xu-yuan1 — APPROVED** [review]

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 4]
```diff
@@ -114,6 +114,11 @@ Following the OpenVINO™ naming convention, the Automatic Device Selection mode
 | |                              | | Indicates the priority for a model.                                |
 | |                              | | IMPORTANT: This property is not fully supported yet.               |
 +--------------------------------+----------------------------------------------------------------------+
+| | `ov::execution_devices`      | | Lists the runtime target devices on which the inferences are being |
+| |                              | | executed.                                                          |
+| |                              | | Examples of returning results could be `CPU_HELP`, `CPU`, `GPU`,   |
```
Comment (wangleis) [inline docs/OV_Runtime_UG/auto_device_selection.md:119]: Please add explanation for CPU_HELP situation. 
 
CPU_HELP as output name will be updated. Please follow up.

Comment (WeldonWangwang) [reply docs/OV_Runtime_UG/auto_device_selection.md]: Added it, please review.

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 5]
```diff
@@ -195,25 +201,45 @@ AUTO will then query all available devices and remove CPU from the candidate lis
...
+
+   .. code-block:: sh
+
+      ov::Any execution_devices = compiled_model.get_property(ov::execution_devices)
+
+.. tab:: Python
+
+   .. code-block:: sh
+
+      execution_devices = compiled_model.get_property("EXECUTION_DEVICES")
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/auto_device_selection.md:220]: put code snippets to `docs/snippets`

Comment (WeldonWangwang) [reply docs/OV_Runtime_UG/auto_device_selection.md]: The example code has been moved to `docs/snippets`

File: docs/snippets/AUTO7.cpp [hunk 6]
```diff
@@ -0,0 +1,18 @@
+#include <openvino/openvino.hpp>
+
+int main() {
```
Comment (ilya-lavrenov) [inline docs/snippets/AUTO7.cpp:3]: don't use `main`, give another name like here https://github.com/openvinotoolkit/openvino/pull/15119

**Review by peterchen-intel — APPROVED** [review]

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 7]
```diff
@@ -114,6 +114,12 @@ Following the OpenVINO™ naming convention, the Automatic Device Selection mode
 | |                              | | Indicates the priority for a model.                                |
 | |                              | | IMPORTANT: This property is not fully supported yet.               |
 +--------------------------------+----------------------------------------------------------------------+
+| | `ov::execution_devices`      | | Lists the runtime target devices on which the inferences are being |
+| |                              | | executed.                                                          |
+| |                              | | Examples of returning results could be `CPU_HELP`(`CPU_HELP` is a  |
+| |                              | | temporary device, indicate that CPU as acceleration at the model   |
```
Comment (xu-yuan1) [inline docs/OV_Runtime_UG/auto_device_selection.md:120]:
```suggestion
| |                              | | temporary device, indicating that CPU is used for acceleration at the model   |
```

**Review by xu-yuan1 — APPROVED** [review]

**Review by wangleis — APPROVED** [review]

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 8]
```diff
@@ -195,25 +201,45 @@ AUTO will then query all available devices and remove CPU from the candidate lis
...
+    .. doxygensnippet:: docs/snippets/ov_auto.py
+       :language: python
+       :fragment: [part7]
+
+@endsphinxdirective
+
+### Performance Hints for AUTO
+The `ov::hint::performance_mode` property enables you to specify a performance option for AUTO to be more efficient for particular use cases. The default hint for AUTO is `LATENCY`.
+
+> **NOTE**: Currently, the `ov::hint` property is supported by CPU and GPU devices only.
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/auto_device_selection.md:227]: we don't have ov::hint property, it's a C++ namespace. Do you mean performance mode?

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 9]
```diff
@@ -195,25 +201,45 @@ AUTO will then query all available devices and remove CPU from the candidate lis
...
 
 #### LATENCY
 This option prioritizes low latency, providing short response time for each inference job. It performs best for tasks where inference is required for a single input image, e.g. a medical analysis of an ultrasound scan image. It also fits the tasks of real-time or nearly real-time applications, such as an industrial robot's response to actions in its environment or obstacle avoidance for autonomous vehicles.
 
+> **NOTE**: If no performance hint is set explicitly, AUTO will set LATENCY for devices that have not set `ov::device::properties`.
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/auto_device_selection.md:232]: not all users understand what `ov::device::properties` means here. Could you please be more specific? E.g. `ov::device::properties(<DEVICE_NAME>, ov::hint::performance_mode(ov::hint::LATENCY))`

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 8 above)
Comment (peterchen-intel) [reply docs/OV_Runtime_UG/auto_device_selection.md:227]: @ilya-lavrenov Yes. `ov::hint` property means `ov::hint::performance_mode` property here, I think we need to remove this **NOTE**, what is your opinion?

Comment (ilya-lavrenov) [reply docs/OV_Runtime_UG/auto_device_selection.md:227]: agree

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 9 above)
Comment (WeldonWangwang) [reply docs/OV_Runtime_UG/auto_device_selection.md]: Added it, please review.

**Review by ilya-lavrenov — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
@kblaszczak-intel should we merge?

**Review by kblaszczak-intel — APPROVED** [review]
I would move the ### Checking Target Runtime Devices towards the bottom of the article, to reflect the order from the property table. But we can do it after merging and porting, not to create any mess ;)

**Pull request merged**